packages/protocol/contracts/psp/AirnodePsp.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "../rrp/interfaces/IAirnodeRrp.sol";
+
+// PSP only supports templates, and not "full subscriptions". Full requests are only more
+// economical for RRP because request-time parameters are not written to storage and
+// emitted as logs. Here, both the template parameters and the request-time parameters
+// are stored in storage, so creating a template and making a "template subscription" would
+// have been equivalent to making a "full subscription". 
+contract AirnodePsp {
+    using ECDSA for bytes32;
+
+    // Note that we do not specify `sponsorWallet`, as it won't be checked in `fulfill()`.
+    // This is because botched fulfillments are not a threat (read more at `fulfill()` comments).
+    // We do not specify `sponsor` because it is returned by the allocator along with the respective
+    // `subscriptionId`. This implies that subscriptions can be served by anyone with
+    // signed data.
+    // The struct below could do without the `parameters` parameter (and we would force the
+    // user to put everything in the template `parameters`). However, RRP is designed around the
+    // reuse of templates and Airnode ABI is not very efficient in encoding parameters, which
+    // is good enough reason to allow PSP to reuse more generic templates with additional
+    // subscription-specific parameters.
+    struct Subscription {
+        bytes32 templateId;
+        address conditionAddress;
```

Comment:
I like that condition and fulfill address can be different - I think that is very important